cratedb_toolkit/util/database.py

```diff
@@ -22,12 +27,23 @@ def __init__(self, dburi: str):
         self.engine = sa.create_engine(self.dburi, echo=False)
         self.connection = self.engine.connect()
 
-    def run_sql(self, sql: str, records: bool = False, ignore: str = None):
+    def run_sql(self, sql: t.Union[str, Path, io.IOBase], records: bool = False, ignore: str = None):
         """
         Run SQL statement, and return results, optionally ignoring exceptions.
         """
+
+        sql_effective: str
+        if isinstance(sql, str):
+            sql_effective = sql
+        elif isinstance(sql, Path):
+            sql_effective = sql.read_text()
+        elif isinstance(sql, io.IOBase):
+            sql_effective = sql.read()
+        else:
+            raise TypeError("SQL statement type must be either string, Path, or IO handle")
```

Comment:
Just curious: couldn't this be done by using function overloads with `@typing.overload()` instead?

Reply:
Hi. I have to admit I never used it. If you think the code can be improved, I will be happy to learn about it.